Week2/todo_app/program.js

```diff
@@ -21,38 +22,89 @@ class TodoModel {
         });
     }
 
-    create(description, callback) {
+    create(description, isCompleted, userId, callback) {
         // Write code and query to create a new TODO item
+        const createTodoItem = `INSERT INTO todo_items (text, is_completed, user_id) VALUES (${description}, ${isCompleted}, ${userId});`
+        this.dbConnection.query(createTodoItem, function(err, results, fields){
+            if(err){
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     update(id, description, callback) {
         // Write code and query to update and existing TODO item
+        const updateDescription = `UPDATE todo_items SET text = ${description} WHERE id = ${id};`
+        this.dbConnection.query(updateDescription, function(err, results, fields){
+            if(err){
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     delete(id, callback) {
         // Write code and query to delete an existing TODO item
+        const deleteTodoItem = `DELETE FROM todo_items WHERE id = ${id};`
+        this.dbConnection.query(deleteTodoItem, function(err, results, fields){
+            if(err){
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
         // Write code and query add a tag to a TODO item
+        const tagTodoItem = `INSERT INTO todo_item_tag (todo_item_id, tag_id) VALUES (${todoItemId}, ${tagId});`
```

Comment:
:+1: 

This assumes the tag already exists - what if it's a new tag that hasn't been used before?